### DISCUSSION TITLE
How can I delete all my devices and the tuya local integration?

### DISCUSSION THREAD

**Discussion by lgander** [make-all/tuya-local#640] [Q&A] [answered]
I can't seem to find a way to do this.
I can disble the devices, but how can I delete them?
I can delete the entities, but the devices are still there.

And finally, how can I completely remove the integration?

Thanks!

**Comment by make-all** [comment, marked as answer]
In Settings / Devices and Services on the Integrations tab, you can select the device you want to delete from inside the Tuya Local box, and on the menu there is a "Delete" option.  Once you've deleted all the devices, you can go ahead and remove the integration.  If you installed via HACS, I think you can remove the integration from the HACS UI.  If you installed manually, then you will need to manually remove the files and restart.

**Reply by lgander** [reply]
Thanks!
